pyaedt/circuit.py

```diff
@@ -1687,3 +1688,404 @@ def import_edb_in_circuit(self, edb_path):
+        for pin in sub.pins:
+            loc = pin.location
+            if loc[0] < center:
+                delta = unit_converter(start + left * 200, input_units="mil", output_units=self.modeler.schematic_units)
+                new_loc = [loc[0] - delta, loc[1]]
+                left += 1
+            else:
+                delta = unit_converter(
+                    start + right * 200, input_units="mil", output_units=self.modeler.schematic_units
+                )
+                new_loc = [loc[0] + delta, loc[1]]
+                right += 1
+            port = self.modeler.components.create_interface_port(name=pin.name, location=new_loc)
+            port.pins[0].connect_to_component(component_pin=pin, use_wire=True)
+            ports.append(port)
+        diff_pairs = []
+        comm_pairs = []
+        if auto_assign_diff_pairs:
+            for pin in ports:
+                pin_name = pin.name.split(separation)
+                if pin_name[-1][-1] == "P":
+                    component = pin_name[pattern.index("component")]
+                    net = pin_name[pattern.index("net")]
+                    for neg_pin in ports:
+                        neg_pin_name = neg_pin.name.split(separation)
+                        component_neg = neg_pin_name[pattern.index("component")]
+
+                        net_neg = neg_pin_name[pattern.index("net")]
+
+                        if component_neg == component and net_neg != net and net_neg[:-1] == net[:-1]:
+                            self.set_differential_pair(
+                                positive_terminal=pin.name,
+                                negative_terminal=neg_pin.name,
+                                common_name=f"COMMON_{component}_{net}",
+                                diff_name=f"{component}_{net}",
+                                common_ref_z=25,
+                                diff_ref_z=100,
+                                active=True,
+                            )
+                            diff_pairs.append(f"{component}_{net}")
+                            comm_pairs.append(f"COMMON_{component}_{net}")
+                            break
+        setup1 = self.create_setup()
+        setup1.props["SweepDefinition"]["Data"] = "LINC {}GHz {}GHz 1001".format(start_frequency, stop_frequency)
+        if analyze:
+            self.analyze()
+        return True, diff_pairs, comm_pairs
+
+    @pyaedt_function_handler()
+    def create_ami_schematic_from_snp(
+        self,
+        touchstone,
+        ibis_ami,
+        component_name,
+        tx_buffer_name,
+        rx_buffer_name,
+        tx_pins,
+        tx_refs,
+        rx_pins,
+        rx_refs,
+        use_ibis_buffer=True,
+        differential=True,
+        bit_pattern=None,
+        unit_interval=None,
+        use_convolution=True,
+        analyze=False,
+        design_name="AMI",
+    ):
+        """Create a new schematic from a touchstone file and automatically setup an IBIS AMI analysis.
```

Comment:
```suggestion
        """Create a schematic from a Touchstone file and automatically set up an IBIS-AMI analysis.
```
Hyphenated "IBIS-AMI" based on a Google search for the term.